tests/resource/sns_topic: Make FIFO topic tests GovCloud compatible

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

SNS FIFO topics are not support in GovCloud.
Relevant acceptance tests are failing:

```
=== RUN   TestAccAWSSNSTopic_Name_FIFOTopic
=== PAUSE TestAccAWSSNSTopic_Name_FIFOTopic
=== CONT  TestAccAWSSNSTopic_Name_FIFOTopic
    resource_aws_sns_topic_test.go:347: Step 1/2 error: Error running apply: exit status 1
        
        Error: error creating SNS Topic (tf-acc-test-3760486360999710358.fifo): InvalidParameter: Invalid parameter: Attributes Reason: Unknown attribute FifoTopic
        	status code: 400, request id: 71ca8514-411a-5b27-97a6-dde529a0e08f
        
          on terraform_plugin_test.tf line 2, in resource "aws_sns_topic" "test":
           2: resource "aws_sns_topic" "test" {
        
        
--- FAIL: TestAccAWSSNSTopic_Name_FIFOTopic (11.80s)
```

### References

* #15828

--- a/aws/provider_test.go
+++ b/aws/provider_test.go
@@ -1067,8 +1067,9 @@ func testAccErrorCheckSkipMessagesContaining(t *testing.T, messages ...string) r
 		}
 
 		for _, message := range messages {
-			if strings.Contains(err.Error(), message) {
-				t.Skipf("skipping test for %s/%s: %s", testAccGetPartition(), testAccGetRegion(), err.Error())
+			errorMessage := err.Error()
+			if strings.Contains(errorMessage, message) {
+				t.Skipf("skipping test for %s/%s: %s", testAccGetPartition(), testAccGetRegion(), errorMessage)
 			}
 		}
 

--- a/aws/resource_aws_sns_topic_subscription_test.go
+++ b/aws/resource_aws_sns_topic_subscription_test.go
@@ -414,7 +414,7 @@ func TestAccAWSSNSTopicSubscription_firehose(t *testing.T) {
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
-		ErrorCheck:   testAccErrorCheckSkipSNS(t),
+		ErrorCheck:   testAccErrorCheck(t, sns.EndpointsID),
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSSNSTopicSubscriptionDestroy,
 		Steps: []resource.TestStep{
@@ -480,13 +480,6 @@ func TestAccAWSSNSTopicSubscription_disappears_topic(t *testing.T) {
 			},
 		},
 	})
-}
-
-// testAccErrorCheckSkipSNS skips SNS tests that have error messages indicating unsupported features
-func testAccErrorCheckSkipSNS(t *testing.T) resource.ErrorCheckFunc {
-	return testAccErrorCheckSkipMessagesContaining(t,
-		"Invalid protocol type: firehose",
-	)
 }
 
 func testAccCheckAWSSNSTopicSubscriptionDestroy(s *terraform.State) error {

--- a/aws/resource_aws_sns_topic_test.go
+++ b/aws/resource_aws_sns_topic_test.go
@@ -18,6 +18,8 @@ import (
 )
 
 func init() {
+	RegisterServiceErrorCheckFunc(sns.EndpointsID, testAccErrorCheckSkipSNS)
+
 	resource.AddTestSweepers("aws_sns_topic", &resource.Sweeper{
 		Name: "aws_sns_topic",
 		F:    testSweepSnsTopics,
@@ -86,6 +88,13 @@ func testSweepSnsTopics(region string) error {
 	}
 
 	return sweeperErrs.ErrorOrNil()
+}
+
+func testAccErrorCheckSkipSNS(t *testing.T) resource.ErrorCheckFunc {
+	return testAccErrorCheckSkipMessagesContaining(t,
+		"Invalid protocol type: firehose",
+		"Unknown attribute FifoTopic",
+	)
 }
 
 func TestAccAWSSNSTopic_basic(t *testing.T) {

